input/musig2.go

```diff
@@ -0,0 +1,228 @@
+package input
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+const (
+	// MuSig2PartialSigSize is the size of a MuSig2 partial signature.
+	// Because a partial signature is just the s value, this corresponds to
+	// the length of a scalar.
+	MuSig2PartialSigSize = 32
+)
+
+// MuSig2SessionID is a type for a session ID that is just a hash of the MuSig2
+// combined key and the local public nonces.
+type MuSig2SessionID = [sha256.Size]byte
+
+// MuSig2Signer is an interface that declares all methods that a MuSig2
+// compatible signer needs to implement.
+type MuSig2Signer interface {
+	// MuSig2CreateSession creates a new MuSig2 signing session using the
+	// local key identified by the key locator. The complete list of all
+	// public keys of all signing parties must be provided, including the
+	// public key of the local signing key. If nonces of other parties are
+	// already known, they can be submitted as well to reduce the number of
+	// method calls necessary later on.
+	MuSig2CreateSession(keychain.KeyLocator, []*btcec.PublicKey,
+		*MuSig2Tweaks, [][musig2.PubNonceSize]byte) (*MuSig2SessionInfo,
```

Comment:
No, we don't make the distinction because with the previous design of the API we needed the session to get our local nonce so we could give it to the other parties. 

The list of other party public nonces is not optional as in dynamic argument list. But you can pass in `nil` if you're the first to create a session and don't know any nonces of other participants yet.

I'll see how this looks once I've incorporated the recent API changes in the context/session. I hope I can add that functionality with minimal changes to the RPC API surface.